### ISSUE TITLE
User Warning thrown when calling AveragePrecision

### ISSUE THREAD

**Issue by nishprabhu** [PyTorchLightning/metrics#475]
## 🐛 Bug

Using the AveragePrecision metric throws the following warning every time it is called:

UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).

### To Reproduce

1. Import AveragePrecision and create an object of this class
2. Call the object by passing in pred and target tensors

#### Code sample

```
import torch
from torchmetrics import AveragePrecision

a = torch.softmax(torch.randn(4, 4), dim=1)
b = torch.randint(0, 2, (4, 4))

ap = AveragePrecision(pos_label=1)
ap(a, b)
```

### Expected behavior
This warning should not be thrown. The recommended way of copying a tensor should be followed.

### Environment

- PyTorch Version: 1.8.2
- OS (e.g., Linux): Ubuntu 18
- How you installed PyTorch (`conda`, `pip`, source): conda
- Build command you used (if compiling from source):
- Python version: 3.8.5
- CUDA/cuDNN version: 11.1/8.0.5
- GPU models and configuration: Quadro RTX 8000
- Any other relevant information:

### Additional context

The problem is arising due to line 156 in torchmetrics/functional/classification/precision_recall_curve.py
The line currently reads:
```
thresholds = tensor(reversed(thresholds[sl]))
```
This problem should be fixed by changing that line to:
```
thresholds = reversed(thresholds[sl]).detach().clone()
```

**Comment by Borda** [comment]
good catch, shall be a simple fix, are you interested in sending a PR? :rabbit:

**Comment by nishprabhu** [comment]
Sure, I'll send a PR. :)
